pkg/proxy/proxier.go

```diff
@@ -80,6 +83,29 @@ func tryConnect(service string, srcAddr net.Addr, protocol string, proxier *Prox
 		outConn, err := net.DialTimeout(protocol, endpoint, retryTimeout*time.Second)
 		if err != nil {
 			glog.Errorf("Dial failed: %v", err)
+			if client != nil {
+				host, err := os.Hostname()
+				if err != nil {
+					glog.Errorf("failed to get hostname: %v", err)
+				}
+				event := &api.Event{
+					InvolvedObject: api.ObjectReference{
+						Kind:      "Service",
+						Name:      service,
+						Namespace: namespace,
+						FieldPath: fmt.Sprintf("pods.%s", endpoint),
+					},
+					Reason:  "CONNECT_FAILED",
+					Message: err.Error(),
+					Source: api.EventSource{
+						Component: "kube-proxy",
+						Host:      host,
+					},
+				}
+				if _, err := client.Events(api.NamespaceDefault).Create(event); err != nil {
```

Comment:
I'm not sure this is event-worthy.  In a dynamic system there will always be things coming up and going down, this is expected.  Also, if/when we move to pure iptables this will be very hard to retain.